interfaces/python/CMakeLists.txt

```diff
@@ -157,16 +179,23 @@ if(HELICS_ENABLE_SWIG AND SWIG_EXECUTABLE)
     endif()
 
     if(NOT USE_MODERNIZED_SWIG)
-        include_directories(${HELICS_Python3_INCLUDE_DIRS})
+        include_directories(${HELICS_Python_INCLUDE_DIRS})
         message(STATUS "Using old SWIG include_directories")
     endif()
 
-    set(CMAKE_SWIG_FLAGS "-py3")
+    if(_py_ver EQUAL 3)
+        set(CMAKE_SWIG_FLAGS "-py3")
+    endif()
 
     if(SWIG_VERSION VERSION_GREATER "4.0.0")
         set(CMAKE_SWIG_FLAGS ${CMAKE_SWIG_FLAGS} "-doxygen")
     endif()
 
+    set(_helicsPython.i_FILE "helicsPython.i")
```

Comment:
I suppose we could rename the .i file to helicsPython3.i and then you could use the _py_ver variable to find it

Reply:
If we're okay with requiring SWIG 3.0.9+ (released in May 2016), we can use the same helicsPython.i and type_maps.i file for both Python 2 and 3 -- there's just one section that would need to use the `SWIGPYTHON_PY3` define.

Reply:
Not sure we can make that a requirement yet. Definitely can for HELICS 3.0 though.

Reply:
Not sure we can make that a requirement yet. Definitely can for HELICS 3.0 though.

Reply:
Not sure we can make that a requirement yet. Definitely can for HELICS 3.0 though.

Reply:
The other option would be to include the codegen for python2 in the repo.  Then we could make the requirement.  But we need to be able to build the python 2 on older systems yet.

Reply:
The SWIG 3.0.9+ requirement would be for building the Python 3 interface with SWIG; any SWIG 3 version would work for building the Python 2 interface since the fallback code if `SWIGPYTHON_PY3` is undefined (always for pre-3.0.9) would be the Python 2 compatible type map.

Reply:
I'm also planning on updating the packaging code to build Python 2 binary distributions for Windows/Linux/macOS that can be installed with pip.

Reply:
I think @kdheepak  should weigh in, but I don't like the idea of adding an additional requirement right now for the 2.4 release.   From the sound of it we can make it pretty easy to workaround so I want to take a closer look after the release, and I do like the idea of having a combined python2 and 3 section of the build.  So we can evaluated it closely for a 2.5 release an definitely the 3.0 release.  But I really hesitate to change it for the 2.4 release.